### PR TITLE
Update SG: add 2025 official holidays

### PR DESCRIPTION
**Pull request by mborsetti**
## Proposed change

Added 2025 official holidays for Singapore

Source: Official Government news release at https://www.mom.gov.sg/newsroom/press-releases/2024/0805-public-holidays-for-2025

Testing source: official dates at https://www.mom.gov.sg/employment-practices/public-holidays

## Type of change

- [ ] New country/market holidays support (thank you!)
- [X] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/pin/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [X] I've followed the [contributing guidelines][contributing-guidelines]
- [] I've run `make pre-commit`, it didn't generate any changes
  **Not working, but Git runs it automatically**
  ```
  make pre-commit
  make: The term 'make' is not recognized as a name of a cmdlet, function, script file, or executable program.
  Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
  ```
- [] I've run `make test`, all tests passed locally
 **Not working, but ran it manually from IDE**
  ```
  make test
  make: The term 'make' is not recognized as a name of a cmdlet, function, script file, or executable program.
  Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
  ```

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/dev/docs/source

### CODE REVIEW

File: holidays/countries/singapore.py [hunk 1]
```diff
@@ -264,8 +269,10 @@ class SingaporeIslamicHolidays(_CustomIslamicHolidays):
         2022: (JUL, 10),
         2023: (JUN, 29),
         2024: (JUN, 17),
+        2025:(JUN, 7,),
```
Comment (arkid15r) [inline holidays/countries/singapore.py:272]: This is going to be flagged by pre-commit
```suggestion
        2025: (JUN, 7),
```

File: holidays/countries/singapore.py [hunk 2]
```diff
@@ -291,6 +298,7 @@ class SingaporeIslamicHolidays(_CustomIslamicHolidays):
         2022: (MAY, 3),
         2023: (APR, 22),
         2024: (APR, 10),
+        2025:(MAR, 31),
```
Comment (arkid15r) [inline holidays/countries/singapore.py:301]:
```suggestion
        2025: (MAR, 31),
```

**Review by arkid15r — COMMENTED** [review]
Hi Mike,
Thanks for keeping this updated!

Could you check the formatting comments before queuing it to the pipeline:

**Comment by mborsetti** [comment]
I guess `pre-commit` did not run automatically.  And when run manually it doesn't do anything:

```
~\Documents\python\git_repos\python-holidays git:[sg_2025]
pre-commit
check python ast.....................................(no files to check)Skipped
check builtin type constructor use...................(no files to check)Skipped
check yaml...........................................(no files to check)Skipped
fix end of files.....................................(no files to check)Skipped
fix python encoding pragma (deprecated)..............(no files to check)Skipped
mixed line ending....................................(no files to check)Skipped
trim trailing whitespace.............................(no files to check)Skipped
ruff.................................................(no files to check)Skipped
ruff-format..........................................(no files to check)Skipped
check license headers................................(no files to check)Skipped
isort................................................(no files to check)Skipped
mypy.................................................(no files to check)Skipped
rstcheck.............................................(no files to check)Skipped
rst ``code`` is two backticks........................(no files to check)Skipped
tox-ini-fmt..........................................(no files to check)Skipped
```

Oh, I see it doesn't even have `black` or similar code formatter -- this was supposed to be a quick contribution, not a belabored one, and for those I rely on those automated tools to do the spacing cleanup for me.

Hopefully I manually caught everything; if not, would you be so kind to fix? Thx

BTW amazing progress -- I didn't have any time to look outside of this file, but love the structure (and hopefully runs much faster too!)

**Review by arkid15r — APPROVED** [review]
LGTM!

Thank you @mborsetti

**Pull request merged**